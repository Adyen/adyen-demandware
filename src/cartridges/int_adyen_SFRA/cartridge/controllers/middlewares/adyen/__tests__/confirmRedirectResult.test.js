/* eslint-disable global-require */
let confirmRedirectResult;
let res;
let req;

beforeEach(() => {
  const { adyen } = require('../../index');
  confirmRedirectResult = adyen.confirmRedirectResult;
  jest.clearAllMocks();

  res = {
    redirect: jest.fn(),
    json: jest.fn(),
  };

  req = {
    querystring: {},
  };
});

afterEach(() => {
  jest.resetModules();
});

describe('Confirm redirectResult', () => {
  it('should have redirectResult', () => {
    const adyenCheckout = require('*/cartridge/scripts/adyenCheckout');
    req.querystring.redirectResult = 'mocked_redirect_result';
    confirmRedirectResult(req, res, jest.fn());
    expect(adyenCheckout.doPaymentDetailsCall.mock.calls).toMatchSnapshot();
  });

  it('should fail with missing redirectResult', () => {
    const adyenCheckout = require('*/cartridge/scripts/adyenCheckout');
    const URLUtils = require('dw/web/URLUtils');
    confirmRedirectResult(req, res, jest.fn());
    expect(adyenCheckout.doPaymentDetailsCall).not.toHaveBeenCalled();
    expect(URLUtils.url.mock.calls[0][0]).toBe('PaymentInstruments-AddPayment');
    expect(URLUtils.url.mock.calls[0][1]).toBe('mocked_error.payment.not.valid');
  });

  test.each(['Authorised', 'Cancelled', 'Error', 'Refused'])(
    'should handle payment confirmation statuses: %p',
    (resultCode) => {
      const adyenCheckout = require('*/cartridge/scripts/adyenCheckout');
      const URLUtils = require('dw/web/URLUtils');

      adyenCheckout.doPaymentDetailsCall.mockImplementation(() => ({
          resultCode: resultCode,
          pspReference: 'mocked_pspReference',
      }));
      req.querystring.redirectResult = 'mocked_redirect_result';
      confirmRedirectResult(req, res, jest.fn());
      expect(res.json.mock.calls[0][0]).toEqual({
          resultCode: resultCode,
          pspReference: 'mocked_pspReference',
      })
      expect(URLUtils.url).not.toHaveBeenCalled();
    },
  );
});
