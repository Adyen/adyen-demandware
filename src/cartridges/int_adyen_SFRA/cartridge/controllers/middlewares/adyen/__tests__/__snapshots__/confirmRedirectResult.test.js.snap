// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm redirectResult should have redirectResult 1`] = `
Array [
  Array [
    Object {
      "details": Object {
        "redirectResult": "mocked_redirect_result",
      },
    },
  ],
]
`;
